modules/transcription-service-amberscript/src/main/java/org/opencastproject/transcription/amberscript/AmberscriptTranscriptionService.java

```diff
@@ -264,56 +303,78 @@ public void activate(ComponentContext cc) {
 
-    amberscriptLangUtil = AmberscriptLangUtil.getInstance();
-    int customMapEntriesCount = 0;
-    Option<String> langCodeMapOpt = OsgiUtil.getOptCfg(cc.getProperties(), LANGUAGE_CODE_MAP);
-    if (langCodeMapOpt.isSome()) {
+    Option<String> speakerMetadataFieldOpt = OsgiUtil.getOptCfg(cc.getProperties(), SPEAKER_METADATA_FIELD);
+    if (speakerMetadataFieldOpt.isSome()) {
       try {
-        String langCodeMapStr = langCodeMapOpt.get();
-        if (langCodeMapStr != null) {
-          for (String mapping : langCodeMapStr.split(",")) {
-            String[] mapEntries = mapping.split(":");
-            amberscriptLangUtil.addCustomMapping(mapEntries[0], mapEntries[1]);
-            customMapEntriesCount += 1;
-          }
-        }
-      } catch (Exception e) {
-        logger.warn("Configuration '{}' is invalid. Using just default mapping.", LANGUAGE_CODE_MAP);
+        speakerMetadataField = SpeakerMetadataField.valueOf(speakerMetadataFieldOpt.get());
+      } catch (IllegalArgumentException e) {
+        logger.warn("Value '{}' is invalid for configuration '{}'. Using default: '{}'.",
+            speakerMetadataFieldOpt.get(), SPEAKER_METADATA_FIELD, speakerMetadataField);
       }
     }
-    logger.info("Language code map was set. Added '{}' additional entries.", customMapEntriesCount);
+    logger.info("Default metadata field for calculating the amount of speakers is set to '{}'.", speakerMetadataField);
+
+    Option<String> transcriptionTypeOpt = OsgiUtil.getOptCfg(cc.getProperties(), TRANSCRIPTIONTYPE);
+    if (transcriptionTypeOpt.isSome()) {
+      transcriptionType = transcriptionTypeOpt.get();
+      logger.info("Default transcription type is set to '{}'.", transcriptionType);
+    } else {
+      logger.info("Default transcription type '{}' will be used.", transcriptionType);
+    }
+
+    Option<String> glossaryOpt = OsgiUtil.getOptCfg(cc.getProperties(), GLOSSARY);
+    if (glossaryOpt.isSome()) {
+      glossary = glossaryOpt.get();
+      logger.info("Default glossary is set to '{}'.", glossary);
+    } else {
+      logger.info("No glossary will be used by default");
+    }
+
+    Option<String> transcriptionStyleOpt = OsgiUtil.getOptCfg(cc.getProperties(), TRANSCRIPTIONSTYLE);
+    if (transcriptionStyleOpt.isSome()) {
+      transcriptionStyle = transcriptionStyleOpt.get();
+      logger.info("Default transcription style is set to '{}'.", transcriptionStyle);
+    } else {
+      logger.info("Default transcription style '{}' will be used.", transcriptionStyle);
+    }
+
+    Option<String> targetLanguageOpt = OsgiUtil.getOptCfg(cc.getProperties(), TARGETLANGUAGE);
+    if (targetLanguageOpt.isSome()) {
+      targetLanguage = targetLanguageOpt.get();
+      logger.info("Default target language is set to '{}'.", targetLanguage);
+    } else {
+      logger.info("Transcriptions won't be translated");
```

Comment:
Hold on now, I read the docs to mean that we're looking for transcriptions in `targetLanguage`, not that Amberscript would be running the transcriptions through some kind of translation tool.  Which is it?

Reply:
`language` is the language Amberscript expects the audio to be in. `targetLanguage` is the language of the captions that you get. If they differ there will be some kind of translation AFAIU, yes. That's what I tried to clarify in 5d05741b1d1fd18c20582e8b5269a99af3491ca6. :thinking: